### PR TITLE
feat: Support jsonc config files

### LINKED ISSUE #198: support deno.jsonc

### PR DESCRIPTION
**Pull request by arnauorriols**
Resolves #198

### CODE REVIEW

File: deps.ts [hunk 1]
```diff
@@ -13,13 +14,15 @@ export {
...
   bold,
+  cyan,
   green,
   magenta,
   red,
   yellow,
 } from "https://deno.land/std@0.170.0/fmt/colors.ts";
 export { parse } from "https://deno.land/std@0.195.0/flags/mod.ts";
 export { TextLineStream } from "https://deno.land/std@0.170.0/streams/text_line_stream.ts";
+export * as JSONC from "https://deno.land/std@0.170.0/encoding/jsonc.ts";
```
Comment (magurotuna) [inline deps.ts:25]: Why not use more recent version? https://deno.land/std@0.208.0/jsonc/mod.ts

Comment (arnauorriols) [reply deps.ts:25]: It's the std version we are using. I'd prefer not to mix versions of the same dependency

File: src/config_file.ts [hunk 2]
```diff
@@ -95,7 +113,9 @@ class ConfigFile {
   }
 
   static fromFileContent(filepath: string, content: string) {
-    const parsedContent = JSON.parse(content);
+    const parsedContent = extname(filepath) === ".jsonc"
+      ? new Object(JSONC.parse(content))
+      : JSON.parse(content);
```
[the remark this answers is not in the document]
Comment (arnauorriols) [reply src/config_file.ts:118]: not really, no. I'll refactor this

File: deps.ts (hunk 1 above)
Comment (magurotuna) [reply deps.ts:25]: In the shown block here there is already mixed version happening (0.195.0 for `parse` and 0.170.0 for `TextLineStream` - but yeah in general I agree that we'd like to stick to the same version. Let's do the version bump in a separate PR.

Comment (magurotuna) [reply deps.ts:25]: https://github.com/denoland/deployctl/issues/216

File: src/config_file.ts [hunk 3]
```diff
@@ -61,20 +73,26 @@ class ConfigFile {
...
-        return false;
+    for (const [key, otherValue] of Object.entries(otherConfigOutput)) {
+      const thisValue = thisConfigOutput[key];
+      if (thisValue !== otherValue) {
+        changes.push({ key, removal: thisValue, addition: otherValue });
       }
     }
-    return true;
+    return changes;
   }
```
[the remark this answers is not in the document]
Comment (arnauorriols) [reply src/config_file.ts:96]: done

**Comment by arnauorriols** [comment]
@magurotuna I've addressed your comments, PTAL.

File: src/config_file.ts [hunk 4]
```diff
@@ -63,28 +75,33 @@ class ConfigFile {
...
+          changes.push({ key, removal: thisValue, addition: otherValue });
         }
       } else if (thisValue !== otherValue) {
-        return false;
+        changes.push({ key, removal: thisValue, addition: otherValue });
       }
     }
-    return true;
+    return changes;
   }
```
[the remark this answers is not in the document]
Comment (arnauorriols) [reply src/config_file.ts:105]: I like using `Object.entries` as I prefer the possible false-negatives resulting from it over the false-positives that we would risk if we explicitly checked each property. The reason being that false-negatives are more visible hence easier to detect.

File: tests/config_file_test/config_file_test.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  changes = config.diff({ project: "bar", entrypoint: "file://src/main.ts" });
+  assertEquals(changes, [
+    { key: "project", removal: "foo", addition: "bar" },
+    {
+      key: "entrypoint",
+      removal: "file://main.ts",
+      addition: "file://src/main.ts",
+    },
+  ]);
+});
```
[the remark this answers is not in the document]
Comment (arnauorriols) [reply tests/config_file_test/config_file_test.ts:33]: Added

**Review by magurotuna — APPROVED** [review]
LGTM👍

**Pull request merged**